### PR TITLE
Fix/scroll top on load

### PR DESCRIPTION
**Pull request by Oliversw**
- Added a higher order component for the router, which checks that the user is on a new page, and if they are, renders the component after scrolling back to the top
- Slightly refactored some variables so that they use the destructuring declaration
- Changed all routes in app.js to use the new routing component

There are still some problems with the linter, but I didn't want to spend too much time on them. Happy for some input about how to fix them though!!

relates #119

### CODE REVIEW

**Comment by hshahwan** [comment]
:+1: 
 just solve conflict

**Review by hshahwan — APPROVED** [review]

**Review by ali-7 — APPROVED** [review]
just need it, propType validation for later

File: client/src/App.js [hunk 1]
```diff
@@ -34,21 +37,24 @@ class App extends Component {
...
+            <ScrollToTopRoute path="/theory" component={Theory} />
+            <ScrollToTopRoute path="/priorities" component={Priorities} />
+            <ScrollToTopRoute
+              path="/priority-tools"
+              component={PrioritiesTools}
+            />
+            <ScrollToTopRoute path="/summary" component={Summary} />
+            <ScrollToTopRoute path="/error404" component={Error404} />
+            <ScrollToTopRoute path="/yourScenario" component={YourScenario} />
+            <RouScrollToTopRoutete path="/activities" component={Activities} />
```
Comment (hshahwan) [inline client/src/App.js]: update 'RouScrollToTopRoutete' to 'ScrollToTopRoute' and Travis will work

**Review by hshahwan — CHANGES_REQUESTED** [review]

File: client/src/App.js (hunk 1 above)
Comment (Oliversw) [reply client/src/App.js]: Just fixed it! Silly mistakes 😅

**Review by hshahwan — APPROVED** [review]

**Comment by hshahwan** [comment]
perfect :+1:

**Pull request merged**